master/buildbot/schedulers/base.py

```diff
@@ -87,12 +87,29 @@ def __init__(self, name, builderNames, properties=None,
 
         # internal variables
         self._change_consumer = None
+        self._enable_consumer = None
         self._change_consumption_lock = defer.DeferredLock()
 
-    # activity handling
+        self.enabled = True
 
+    # activity handling
+    @defer.inlineCallbacks
     def activate(self):
-        return defer.succeed(None)
+        # even if we aren't called via _activityPoll(), at this point we
+        # need to ensure the service id is set correctly
+        if self.serviceid is None:
+            self.serviceid = yield self._getServiceId()
+            assert self.serviceid is not None
+
+        schedulerData = yield self._getScheduler(self.serviceid)
+
+        if schedulerData:
+            self.enabled = schedulerData['enabled']
+
+        yield self.startConsumingEnabledChanges()
```

Comment:
changes is a thing in builbot its about changes to SCM. here we are watching events.